core/src/main/java/org/apache/carbondata/core/indexstore/ExtendedBlockletWrapperContainer.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.indexstore;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.carbondata.common.logging.LogServiceFactory;
+import org.apache.carbondata.core.metadata.schema.table.Writable;
+import org.apache.carbondata.core.util.CarbonProperties;
+import org.apache.carbondata.core.util.CarbonThreadFactory;
+
+import org.apache.log4j.Logger;
+
+/**
+ * below class will be used to send split information from index driver to
+ * main driver.
+ * Main driver will Deserialize the extended blocklet object and get the split
+ * to run the query
+ */
+public class ExtendedBlockletWrapperContainer implements Writable {
+
+  private static final Logger LOGGER =
+      LogServiceFactory.getLogService(ExtendedBlockletWrapperContainer.class.getName());
+
+  private ExtendedBlockletWrapper[] extendedBlockletWrappers;
+
+  public ExtendedBlockletWrapperContainer() {
+
+  }
+
+  public ExtendedBlockletWrapperContainer(ExtendedBlockletWrapper[] extendedBlockletWrappers) {
+    this.extendedBlockletWrappers = extendedBlockletWrappers;
+  }
+
+  public List<ExtendedBlocklet> getExtendedBlockets(String tablePath, String queryId) {
+    int numOfThreads = CarbonProperties.getNumOfThreadsForPruning();
+    ExecutorService executorService = Executors
+        .newFixedThreadPool(numOfThreads, new CarbonThreadFactory("SplitDeseralizerPool", true));
+    int numberOfWrapperPerThread = extendedBlockletWrappers.length / numOfThreads;
+    int leftOver = extendedBlockletWrappers.length % numOfThreads;
+    int[] split = null;
+    if (numberOfWrapperPerThread > 0) {
+      split = new int[numOfThreads];
+    } else {
+      split = new int[1];
+    }
+    Arrays.fill(split, numberOfWrapperPerThread);
+    for (int i = 0; i < leftOver; i++) {
+      split[i] += 1;
```

Comment:
Consider the following example:
`numOfThreads` = 4
`extendedBlockletWrappers.length` = 3
`numberOfWrapperPerThread` = 3/4 --> 0
`leftOver` = 3%4 --> 3
split will be an int[1]
In this case this for loop will throw `ArrayIndexOutOfBoundException`. Kindly check once this scenario

Reply:
@manishgupta88 thanks for pointing this out, yaa array size will be left over size

Reply:
fixed